action.yml

```diff
@@ -115,10 +115,14 @@ runs:
           stack_arguments+=( --pedantic )
         fi
 
-        if [[ "${{ inputs.stack-yaml }}" == 'stack-nightly.yaml' ]]; then
-          if [[ "${stack_arguments[*]}" != *'--resolver'* ]]; then
-            stack_arguments+=( --resolver nightly )
-          fi
+        has_resolver() {
+          grep -Fq -- '--resolver' <<'EOM'
+        ${{ inputs.stack-arguments }}
+        EOM
```

Comment:
I'm assuming this is intentional indentation to avoid extra white space.

Reply:
Yeah, a shell heredoc can't have any indentation. Peeling back the Yaml, this is:

```sh
has_resolver() {
  grep -Fq -- '--resolver' <<'EOM'
some potential
content
here
EOM
}
```